doc/source/whatsnew/v0.23.2.txt

```diff
@@ -83,4 +83,5 @@ Bug Fixes
 **Other**
 
 - Bug in :meth:`Series.nlargest` for signed and unsigned integer dtypes when the minimum value is present (:issue:`21426`)
--
+- Bug in :class:`_Concatenator` should not sort Ordered Dictionaries (:issue:`21510`)
```

Comment:
Can you come up with some less technical? This is too focused on the technical aspect of the change, but you really should be writing something that a casual user could read and understand in the whatsnew

Reply:
Agreed.  I would just say that the bug was causing `pd.concat` to not respect order present in an ordered dictionary object (something along those lines).

Reply:
Please, check my last commit.

Reply:
Getting closer but you should remove any reference to _Concatenator as it is a private class. I would be fine with almost a copy/paste of what @gfyoung suggested above as well